CMakeLists.txt

```diff
@@ -20,25 +22,32 @@ add_compile_options("-Wno-format-security")
 #add_compile_options("-Wstrict-overflow=5")
 #add_compile_options("-Wdisabled-optimization")
 
-enable_testing()
+if (ENABLE_TESTS)
+  enable_testing()
+endif()
 
-if (OFFLINE_BUILDS)
-  include(ExternalProject)
-  ExternalProject_Add(bcc
-    GIT_REPOSITORY https://github.com/iovisor/bcc
-    STEP_TARGETS build update
-    EXCLUDE_FROM_ALL 1
-    UPDATE_DISCONNECTED 1
-    BUILD_COMMAND ${CMAKE_COMMAND} --build . --target bcc-static
-    )
+if (SYSTEM_BCC_LIBRARY)
+  find_package(LibBcc REQUIRED)
+  include_directories(${LIBBCC_INCLUDE_DIRS})
 else()
-  include(ExternalProject)
-  ExternalProject_Add(bcc
-    GIT_REPOSITORY https://github.com/iovisor/bcc
-    STEP_TARGETS build update
-    EXCLUDE_FROM_ALL 1
-    BUILD_COMMAND ${CMAKE_COMMAND} --build . --target bcc-static
-    )
+  if (OFFLINE_BUILDS)
+    include(ExternalProject)
+    ExternalProject_Add(bcc
+      GIT_REPOSITORY https://github.com/iovisor/bcc
```

Comment:
While we're changing this, what if we replaced this (since you can't really do an "offline build" that requires a git pull) with `SOURCE_DIR ${BCC_SOURCE_DIR}`.

We also should add `CMAKE_ARGS -DLUAJIT_INCLUDE_DIR=${LUAJIT_INCLUDE_DIR}` to both `ExternalProject_Add`s because otherwise builds fail (if you have `luajit-devel`). Maybe this should be done automatically with `find_package` but I don't know enough about CMake to know how to get that to work.